### DISCUSSION TITLE
silence dns response comments?

### DISCUSSION THREAD

**Discussion by hyder365** [DNSCrypt/dnscrypt-proxy#2466] [Q&A]
Is it possible to disable these additional comments dnscrypt-proxy is handing out? @jedisct1 

$ host example.com
example.com has address 93.184.216.34
example.com host information "AAAA queries have been locally blocked by dnscrypt-proxy" "Set block_ipv6 to false to disable that feature"

$ host doubleclick.net
doubleclick.net host information "This query has been locally blocked" "by dnscrypt-proxy"

**Comment by jedisct1** [comment]
No, but why?

These are very useful in order to understand why something doesn't work as expected.

Incidentally, that very message helped me last week to understand why I couldn't reach an IPv6-only website, after initially thinking it was a DNS configuration error on their side.

**Comment by hyder365** [comment]
This seems like a misfeature. I don't want the users to know exactly what software is running on the host, similar to unbound's "hide identity" option.

It would be better in my case if the users did not know why something was "wrong" because nothing is actually wrong in this setup.

Would you consider an opt-in or opt-out config option so users can decide?

**Reply by lifenjoiner** [reply]
Maximum anonymity?

**Comment by jedisct1** [comment]
Using open source software is not a shame.

**Comment by lifenjoiner** [comment]
I guess you want:
https://github.com/DNSCrypt/dnscrypt-proxy/blob/d48c811ea98d28975f36eb36f68265a58338e166/dnscrypt-proxy/example-dnscrypt-proxy.toml#L141-L146

<br>

But, `block_ipv6` doesn't follow this. Don't know it is a negligence or intentional.
https://github.com/DNSCrypt/dnscrypt-proxy/blob/d48c811ea98d28975f36eb36f68265a58338e166/dnscrypt-proxy/plugin_block_ipv6.go#L37-L44

**Comment by hyder365** [comment]
@lifenjoiner Thank you for the helpful reply. That's definitely an improvement! This does seem like a bug if it's missing for the IPv6 block function.

**Comment by hyder365** [comment]
blocked_query_response = 'a:0.0.0.0'
block_ipv6 = true

This combination gives me two unneeded lines of output on a blocked domain:

$ host doubleclick.net
doubleclick.net has address 0.0.0.0
doubleclick.net has IPv6 address ::ffff:0.0.0.0
doubleclick.net host information "This query has been locally blocked" "by dnscrypt-proxy"

I think the second one is definitely a bug too. There shouldn't be any v6 response.
